packages/cli/src/lib/cmds/play.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import {Config, GradleWrapper, JdkHelper, AndroidSdkTools,ConsoleLog, Log, GooglePlay, TwaManifest} from '@bubblewrap/core';
+import { Track } from '@bubblewrap/core/dist/lib/GooglePlay';
+import * as fs from 'fs';
+import * as path from 'path';
+import { ParsedArgs } from 'minimist';
+import { TWA_MANIFEST_FILE_NAME } from '../constants';
+import {Prompt, InquirerPrompt} from '../Prompt';
+
+class Play {
+  constructor(
+    private args: ParsedArgs,
+    private config: Config,
+    private androidSdkTools: AndroidSdkTools,
+    private googlePlay: GooglePlay,
+  ){}
+
+  async bootstrapPlay(): Promise<void> {
+    await this.googlePlay.initPlay();
+  }
+
+  async getLargestVersion(): Promise<void> { 
+    // TODO(nohe427): This doesn't exist in the Gradle Play Plugin. Might be wortwhile to build
+      // small integration to do this by getting the service account file?
+  }
+
+  private isInAvailableTracks(userSpecifiedTrack: string): Boolean {
+    const track: string = (userSpecifiedTrack).toLowerCase();
+    const selectedTrack: Track = (<any>Track)[track];
+    if(selectedTrack == (null || undefined)) {
+      return false; // Should probably spit out an error message that track needs to be in [x,y,z]
+    }
+    return true;
+  }
+
+  // bubblewrap play --publish="Internal"
+  async publish(): Promise<void> {
+    //if (this.args.publish) { // This argument should be validated in run()
+    //}
+    //Validate that the publish value is listed in the available Tracks.
+    const userSelectedTrack = this.args.publish.toLowerCase() || "internal"; // If no value was supplied with publish we make it internal.
+    if(!this.isInAvailableTracks(userSelectedTrack)) {
+      return; // Throw error message?
+    }
+    if (this.args.appBundleLocation) {
+      // Check this is a directory that contains our specified file name.
+    }
+
+    
+      // Make tmp directory copy file over signed APK then cleanup.
+      // await this.googlePlay.publishBundle(Track.Internal,)
+  }
+
+  private validServiceAccountJsonFile(path: string | undefined): Boolean { // Return an error or boolean? Log a message?
```

Comment:
Can we use `string?` instead of `string | undefined`? (I can never remember this myself.)

Reply:
I do not believe so. Tested and it made the compiler unhappy